### PR TITLE
fix: rewrite LPVS documentation and add Javadoc comments to controller

### PR DESCRIPTION
**Pull request by o-kopysov**
# Description

The project documentation and build instructions have been updated, and Javadoc has been added for the controller.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Code cleanup/refactoring
- [X] Documentation update
- [X] This change requires a documentation update
- [ ] CI system update
- [ ] Test Coverage update

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,107 +1,84 @@
...
+
+8. Commit your changes with a descriptive commit message. Make sure to mention the issue number with `#XXX` so that GitHub will automatically link your commit with the issue:
+   ```sh
+   git add path/to/my/file
+   git commit -m "A brief description of this change which fixes #42" --signoff
+   ```
+
+9. Pull the latest LPVS code from upstream into your branch:
+   ```sh
+   git pull upstream main
```
Comment (tiokim) [inline .github/CONTRIBUTING.md:61]: Code was fetched by 4, what about using `rebase` instead?

```suggestion
   git rebase upstream/main
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,107 +1,84 @@
...
+   ```
+
+10. Push your code to your forked repository:
+   ```sh
```
Comment (tiokim) [inline .github/CONTRIBUTING.md:65]: I don't think the indentation is applied. Can you check?

File: README.md [hunk 3]
```diff
@@ -10,161 +10,196 @@
...
-    - the same passphrase must be saved in `github.secret` of LPVS backend `application.properties` or `docker-compose.yml` files 
-    - select `Let me select individual events` -> `Pull requests` (make sure that only `Pull requests` is selected)
-    - make it `Active`
-    - press `Add Webhook`
+2. Configure the webhook in your GitHub repository settings:
+   - Go to `Settings` -> `Webhooks`.
+   - Click on `Add webhook`.
+   - Fill in the `Payload URL` with: `http://<IP where LPVS is running>:7896/webhooks`.
+     > If you're using ngrok, the `Payload URL` should be like `https://50be-62-205-136-206.ngrok-free.app/webhooks`.
+     - Install ngrok from [here](https://dashboard.ngrok.com/get-started) (follow steps 1 and 2).
```
Comment (tiokim) [inline README.md:63]: Your link requires the login, so I suggest the other link.

```suggestion
     - Install ngrok from [here](https://ngrok.com/docs/getting-started/#step-2-install-the-ngrok-agent) (follow steps 1 and 2).
```

**Review by tiokim — COMMENTED** [review]
Thanks for the contribution. Please take a look at my review comments!

File: .github/CONTRIBUTING.md (hunk 2 above)
Comment (o-kopysov) [reply .github/CONTRIBUTING.md:65]: Agree with your opinion.

**Comment by o-kopysov** [comment]
@tiokim  I agree with all your comments. Please check the updated files.

**Comment by o-konoval** [comment]
I followed instruction in README.md for title "How to Build and Run LPVS from Source Code", and in this instruction have an error: 
mysql -u[username] -p[password] < src/main/resources/database_dump.sql 

$ mysql -uroot -plpvsroot < src/main/resources/database_dump.sql
mysql: [Warning] Using a password on the command line interface can be insecure.
ERROR 1698 (28000): Access denied for user 'root'@'localhost'

But when used this instruction with sudo it executed without errors:
$ sudo mysql -uroot -plpvsroot < src/main/resources/database_dump.sql
mysql: [Warning] Using a password on the command line interface can be insecure.
$ 

Please check it if possible.

**Comment by o-kopysov** [comment]
> I followed instruction in README.md for title "How to Build and Run LPVS from Source Code", and in this instruction have an error: mysql -u[username] -p[password] < src/main/resources/database_dump.sql
> 
> $ mysql -uroot -plpvsroot < src/main/resources/database_dump.sql mysql: [Warning] Using a password on the command line interface can be insecure. ERROR 1698 (28000): Access denied for user 'root'@'localhost'
> 
> But when used this instruction with sudo it executed without errors: $ sudo mysql -uroot -plpvsroot < src/main/resources/database_dump.sql mysql: [Warning] Using a password on the command line interface can be insecure. $
> 
> Please check it if possible.

I think that sudo is dependant on your OS configuration. 
@tdrozdovsky could you please check on your side?

**Comment by o-konoval** [comment]
> > I followed instruction in README.md for title "How to Build and Run LPVS from Source Code", and in this instruction have an error: mysql -u[username] -p[password] < src/main/resources/database_dump.sql
> > $ mysql -uroot -plpvsroot < src/main/resources/database_dump.sql mysql: [Warning] Using a password on the command line interface can be insecure. ERROR 1698 (28000): Access denied for user 'root'@'localhost'
> > But when used this instruction with sudo it executed without errors: $ sudo mysql -uroot -plpvsroot < src/main/resources/database_dump.sql mysql: [Warning] Using a password on the command line interface can be insecure. $
> > Please check it if possible.
> 
> I think that sudo is dependant on your OS configuration. @tdrozdovsky could you please check on your side?

In above instruction mysql run with sudo too: 
2.2 Open Mysql command line interface
sudo mysql

I checked just run without sudo :
$ mysql
ERROR 1045 (28000): Access denied for user 'qa'@'localhost' (using password: NO)

**Comment by o-konoval** [comment]
> 

Update: checked the command (mysql -uroot -plpvsroot < src/main/resources/database_dump.sql) 
on other OS -Ubuntu 18 (previously it was Ubuntu 22) and the command successfully executed:
$ mysql -uusername -ppassword < src/main/resources/database_dump.sql
mysql: [Warning] Using a password on the command line interface can be insecure.
$
So @o-kopysov was right about my comment. 
Anyway it is not critical but may be add as tip - if the command was not successful please use sudo before the command.

**Comment by o-konoval** [comment]
Please consider -may be added 2 tips for non-advanced users in paragraphs?

1.< How to Build and Run LPVS from Source Code  
1. Build Prerequisites>:

If not installed pip3 it may be installed by command:
sudo apt install python3-pip


3.<4. Build LPVS Application with Maven and Run it>:

The application needs installed Java 11 and Maven for build&run. 
It may be installed by following commands:
sudo apt-get install openjdk-11-jdk
sudo apt install maven

**Review by MoonkiHong — APPROVED** [review]
LGTM.

**Review by tiokim — APPROVED** [review]
LGTM

**Comment by o-kopysov** [comment]
> Please consider -may be added 3 tips for non-advanced users in paragraphs?
> 
> 1.< How to Build and Run LPVS from Source Code
> 
> 1. Build Prerequisites>:
> 
> If not installed pip3 it may be installed by command: sudo apt install python3-pip
> 
> or equivalent for non-Debian OS.
> 
> 2.<4. Build LPVS Application with Maven and Run it>:
> 
> The application needs installed Java 11 and Maven for build&run. It may be installed by following commands: sudo apt-get install openjdk-11-jdk sudo apt install maven
> 
> or equivalent for non-Debian OS.
> 
> 3. <MySQL server installed locally. Install it using the command:
> 
> sudo apt install mysql-server>:
> 
> or equivalent for non-Debian OS.

Item 3 is already presented in the guide. 
Items 1 and 2 are trivial and can be excluded from the guide to simplify reading.

**Pull request merged**